### ISSUE TITLE
Add untracked function 

### ISSUE THREAD

**Issue by dy** [kethan/ulive#2]
Hi @kethan !

Nice philosophy, smallest size so far!
I'd suggest removing `o` / `memo` and adding `untacked` function (since you've removed `peek`) - all it needs is to run effect in zero context (example for usignal):

```js
const untracked = (fn, prev) => (prev = current, current = null, fn(), current = prev)
```

Looking forward, thanks!
I would integrate it in [sprae](https://github.com/dy/sprae/tree/v9) as direct dependency

**Comment by kethan** [comment]
Nice that was a genius attempt. 
I need that o and memo to support fn like signal like in sinuous, voby and solid.js.

Here is the updated code which will be updated later sometime.

```js
let current;
export let o = (val, listeners = new Set(), f) => {
    f = (...next) => {
      if (!next.length) {
        current?.deps.push(listeners.add(current));
        return val;
      }
      if (val !== next[0]) {
        val = next[0];
        for (let cb of listeners) {
          cb && cb(val);
        }
      }
    };
    f.peek = f.toJSON = f.then = f.toString = f.valueOf = () => val;
    return f;
  },
  signal = (val, n = o(val)) => ({
    get value() {
      return n();
    },
    set value(next) {
      n(next);
    },
    ...n,
  }),
  effect = (fn, teardown, run, deps) => (
    run = (prev) => {
      if (teardown?.call) teardown();
      prev = current, current = run;
      try {
        teardown = fn();
      }
      finally {
        current = prev;
      }
    },
      deps = run.deps = [],
      run(),
      (dep) => {
        while (dep = deps.pop()) dep.delete(run);
      }
  ),
  memo = (fn, s = o(), e) => ((e ||= effect(() => s(fn()))), s.peek),
  computed = (fn, s = signal(), c, e) => (
    (c = {
      get value() {
        e ||= effect(() => (s.value = fn()));
        return s.value;
      },
    }), c
  ),
  untracked = (
    fn,
    prev,
    val,
  ) => (prev = current, current = null, val = fn(), current = prev, val);

```

**Comment by dy** [comment]
I think there's 2 points to that code.
1. `computed` should better have signal-compatibility `c.toJSON = c.then = c.toString = c.valueOf = () => c.value`,
2. `o` in your case makes `valueOf` not subscribable, same as `.peek`: `f.valueOf = () => val` - it should better be `f.valueOf = f`

**Comment by kethan** [comment]
Am outside if possible can you modify the above code and send it so I can check

**Comment by dy** [comment]
this
```js
let current;
export let o = (val, listeners = new Set(), f) => (
  f = (...next) => {
    if (!next.length) {
      current?.deps.push(listeners.add(current));
      return val;
    }
    if (val !== next[0]) {
      val = next[0];
      for (let cb of listeners) {
        cb && cb(val);
      }
    }
  },
  f.peek = () => val,
  // NOTE: unfortunately can't do `= f` here, since `.toJSON` receives string argument turning get into set
  f.toJSON = f.then = f.toString = f.valueOf = () => f(),
  f
),
  signal = (val, n = o(val)) => ({
    get value() {
      return n();
    },
    set value(next) {
      n(next);
    },
    ...n,
  }),
  effect = (fn, teardown, run, deps) => (
    run = (prev) => {
      if (teardown?.call) teardown();
      prev = current, current = run;
      try {
        teardown = fn();
      }
      finally {
        current = prev;
      }
    },
    deps = run.deps = [],
    run(),
    (dep) => {
      while (dep = deps.pop()) dep.delete(run);
    }
  ),
  memo = (fn, s = o(), e) => ((e ||= effect(() => s(fn()))), s.peek),
  computed = (fn, s = signal(), c, e) => (
    (c = {
      get value() {
        e ||= effect(() => (s.value = fn()));
        return s.value;
      },
    }),
    c.peek = c.toJSON = c.then = c.toString = c.valueOf = () => c.value,
    c
  ),
  untracked = (
    fn,
    prev,
    val,
  ) => (prev = current, current = null, val = fn(), current = prev, val);

```

**Comment by dy** [comment]
Fixed by #3
